x-pack/plugins/fleet/server/config.ts

```diff
@@ -201,64 +201,68 @@ export const config: PluginConfigDescriptor = {
         defaultValue: () => [],
       }),
 
-      internal: schema.maybe(
-        schema.object({
-          disableILMPolicies: schema.boolean({
-            defaultValue: false,
-          }),
-          fleetServerStandalone: schema.boolean({
-            defaultValue: false,
-          }),
-          onlyAllowAgentUpgradeToKnownVersions: schema.boolean({
-            defaultValue: false,
-          }),
-          activeAgentsSoftLimit: schema.maybe(
-            schema.number({
-              min: 0,
-            })
-          ),
-          retrySetupOnBoot: schema.boolean({ defaultValue: false }),
-          registry: schema.object(
-            {
-              kibanaVersionCheckEnabled: schema.boolean({ defaultValue: true }),
-              excludePackages: schema.arrayOf(schema.string(), { defaultValue: [] }),
-              spec: schema.object(
-                {
-                  min: schema.maybe(schema.string()),
-                  max: schema.string({ defaultValue: REGISTRY_SPEC_MAX_VERSION }),
-                },
-                {
-                  defaultValue: {
-                    max: REGISTRY_SPEC_MAX_VERSION,
-                  },
-                }
-              ),
-              capabilities: schema.arrayOf(
-                schema.oneOf([
-                  // See package-spec for the list of available capiblities https://github.com/elastic/package-spec/blob/dcc37b652690f8a2bca9cf8a12fc28fd015730a0/spec/integration/manifest.spec.yml#L113
-                  schema.literal('apm'),
-                  schema.literal('enterprise_search'),
-                  schema.literal('observability'),
-                  schema.literal('security'),
-                  schema.literal('serverless_search'),
-                  schema.literal('uptime'),
-                ]),
-                { defaultValue: [] }
-              ),
-            },
-            {
-              defaultValue: {
-                kibanaVersionCheckEnabled: true,
-                capabilities: [],
-                excludePackages: [],
-                spec: {
+      internal: schema.object({
```

Comment:
made this schema object non-optional (removed `schema.maybe` wrapper), because the default values do not populate when it is optional, i.e. `kibanaVersionCheckEnabled` is `undefined` instead of `false`